### PR TITLE
Validate ProwJobs using CustomResourceValidation

### PR DESCRIPTION
**Pull request by nikhita**
Hold until https://github.com/kubernetes/test-infra/pull/6196 merges.

/cc kargakis sttts

### CODE REVIEW

File: cluster/ci/config/prow/prow_crd.yaml [hunk 1]
```diff
@@ -13,3 +13,41 @@ spec:
...
+              type: string
+              enum:
+              - "triggered"
+              - "pending"
+              - "success"
+              - "failure"
+              - "aborted"
+              - "error"
+          anyOf:
+          - not:
```
Comment (sttts) [inline cluster/ci/config/prow/prow_crd.yaml]: why this anyOf and not? If `state` is given, `completionTime` is required?

Comment (kargakis) [reply cluster/ci/config/prow/prow_crd.yaml]: We want completionTime to be set when state is terminal (success, failure, error, aborted).

Comment (sttts) [reply cluster/ci/config/prow/prow_crd.yaml]: sounds fine.

**Comment by kargakis** [comment]
thanks @nikhita

**Pull request merged**